src/auth/googleauth.ts

```diff
@@ -277,6 +284,32 @@ export class GoogleAuth {
 
     let credential: JSONClient | null;
     let projectId: string | null;
+
+    // Check the existence of GOOGLE API key.
+    if (!this.apiKey) {
+      if (
+        process.env['GOOGLE_API_KEY'] &&
+        process.env['GOOGLE_APPLICATION_CREDENTIALS']
+      ) {
+        throw new Error(
+          'GOOGLE_API_KEY and GOOGLE_APPLICATION_CREDENTIALS environment variables are mutually exclusive'
```

Comment:
After discussion with Silvano, Andy and Brent, we decide to make GOOGLE_API_KEY and GOOGLE_APPLICATION_CREDENTIALS mutually exclusive across all languages since they are different in nature and hard to implement in Java/C# auth libraries.

GOOGLE_API_KEY is a new environment variable so it shouldn't break anyone.